airflow/models/dataset.py

```diff
@@ -124,6 +128,15 @@ def from_public(cls, obj: DatasetAlias) -> DatasetAliasModel:
     def __repr__(self):
         return f"{self.__class__.__name__}(name={self.name!r})"
 
+    def __hash__(self):
+        return hash(self.name)
+
+    def __eq__(self, other):
+        if isinstance(other, (self.__class__, DatasetAlias)):
+            return self.name == other.name
+        else:
+            return NotImplemented
```

Comment:
Not sure what I feel about implementing these on a SQLAlchemy class. Probably doable? We should find some resource on if these are OK.

Or we could instead just use a dict instead (using the name as key).

Reply:
Not sure whether it's ideal, but it's at least ok I think. DatasetModel is doing the same thing

Reply:
I tried to trace the SQLAlchemy code a bit. Didn't find special handling for `__eq__`